src/main/java/org/jenkins/tools/test/hook/WorkflowCpsHook.java

```diff
@@ -23,17 +24,17 @@ protected String getPluginFolderName(UpdateSite.Plugin currentPlugin){
 
     @Override
     public boolean check(Map<String, Object> info) {
-        return isMultiModuleVersionOfWorkflowCps(info);
-    }
-
-    public static boolean isMultiModuleVersionOfWorkflowCps(Map<String, Object> info) {
+        PomData data = (PomData) info.get("pomData");
         UpdateSite.Plugin plugin = info.get("plugin") != null ? (UpdateSite.Plugin) info.get("plugin") : null;
-        if (plugin != null && plugin.name.equals("workflow-cps") && plugin.version != null) {
+        if (plugin != null && plugin.version != null) {
             VersionNumber pluginVersion = new VersionNumber(plugin.version);
             // 2803 was the final release before it became a multi-module project.
             // The history of groovy-cps history was merged into the repo, so the first multi-module release will be a little over 3500.
             VersionNumber multiModuleSince = new VersionNumber("3500");
-            return pluginVersion.isNewerThan(multiModuleSince);
+            return "org.jenkins-ci.plugins.workflow".equals(data.groupId)
+                    && "workflow-cps".equals(data.artifactId)
+                    && pluginVersion.isNewerThan(multiModuleSince)
+                    && "hpi".equals(data.getPackaging());
```

Comment:
Following the same convention used elsewhere by checking the group ID, artifact ID, and packaging type. In this case we also need to check the version to see if it is one where the plugin is single-module or multi-module.